Feature/guardians recovery

7flash opened the pull request:
### 🗃 Github Issue Or Exmplanation for this PR. (What is it supposed to do and Why is needed)

Enter Issue number here

### ✅ Checklist

- [ ] Github issue details are up to date for people to QA.
- [ ] I have tested all my changes.

### 🕵️‍♂️ Notes for Code Reviewer

_Technical explanations to help your peers review your work - including environments, credentials, etc._

### 🙈 Screenshots

_For all UI changes_

| description 1 | description 2 |
| --- | --- |
| img1 | img2 |

### 👯‍♀️ Paired with

_@github-handle or "nobody" if you did not pair._

7flash committed:
add guardians screen

7flash committed:
add guardian calls

7flash committed:
setup permission wip

7flash committed:
update guardian permission

7flash committed:
claim account transaction

n13 committed:
change auth on claimRecoveredAccount to application auth

n13 committed:
Merge branch 'master' into feature/guardians

n13 committed:
add import

n13 committed:
remove debug code

7flash committed:
allow to recover account while onboarding

gguijarro-c-chwy reviewed (changes requested):
No PR template completed

7flash committed:
recover account

7flash committed:
Merge branch 'master' into feature/guardians-recovery

7flash committed:
fix warnings

n13 committed:
Merge branch 'master' into feature/guardians-recovery

n13 committed:
Merge branch 'feature/guardians-recovery' of https://github.com/JoinSEEDS/seeds_light_wallet into feature/guardians-recovery

n13 committed:
fix retrieve guardians

n13 committed:
fix layout, enable guardians for all

7flash committed:
join process recovery

7flash committed:
fix - save private key when account created

7flash committed:
initialize bloc correctly and ignore foundNoLink event

7flash committed:
remove foundNoLink event

7flash committed:
handle recovers null

7flash committed:
handle empty guardians map

7flash committed:
continue recovery when requested

7flash committed:
required guardians length

gguijarro-c-chwy commented on lib/features/account/account_generator_service.dart:
```diff
@@ -2,7 +2,6 @@ import 'package:flutter/foundation.dart';
 import 'package:seeds/providers/services/http_service.dart';
```
are most or all the changes in this file formatting?

gguijarro-c-chwy commented on lib/providers/services/http_service.dart:
```diff
@@ -52,14 +110,43 @@ class HttpService {
+    final String accountPermissionsURL = "$baseURL/v1/chain/get_account";
+
+    String request = '{ "account_name": $userAccount}';
+    Map<String, String> headers = {"Content-type": "application/json"};
```
these headers are declared in multiple places. Can we create a global variable for them?

gguijarro-c-chwy commented on lib/screens/app/ecosystem/guardians/guardians.dart:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  void initGuardians() {
+    List<String> guardians = [
```
3 guardians? 
What if the user has 5? 
Or only 3?

gguijarro-c-chwy commented on lib/screens/app/profile/profile.dart:
```diff
@@ -418,7 +418,7 @@ class _ProfileState extends State<Profile> {
 
   Widget _guardiansView() {
-    if (FirebaseRemoteConfigService().featureFlagGuardiansEnabled) {
+    if (true || FirebaseRemoteConfigService().featureFlagGuardiansEnabled) {
```
no. 
Hard coded true here is no bueno.

We are controlling this feature using the feature flag

gguijarro-c-chwy commented on lib/screens/onboarding/join_process/bloc.dart:
```diff
@@ -0,0 +1,243 @@
+
+    print('onEvent ' + event.toString());
+
+    if (event is FoundInviteLink) {
```
can we use a switch here?

gguijarro-c-chwy commented on lib/screens/onboarding/join_process/bloc.dart:
```diff
@@ -0,0 +1,243 @@
+  Stream<OnboardingState> mapEventToState(OnboardingEvent event) async* {
+    print('mapEventToState ' + event.toString());
+
+    if (event is FoundInviteLink) {
```
switch ?

7flash replied:
I don't think **switch** statement can be used with **is** operator?

gguijarro-c-chwy commented on lib/screens/onboarding/join_process/events.dart:
```diff
@@ -0,0 +1,76 @@
+abstract class OnboardingEvent {
```
👍

gguijarro-c-chwy commented on lib/screens/onboarding/join_process/export.dart:
```diff
@@ -0,0 +1,26 @@
+export 'package:eosdart_ecc/eosdart_ecc.dart';
```
what is this file used for?

gguijarro-c-chwy commented on lib/screens/onboarding/join_process/screens/show_onboarding_choice.dart:
```diff
@@ -53,17 +55,20 @@ class ShowOnboardingChoice extends StatelessWidget {
+                    ..onTap = () => safeLaunch(
+                        'https://www.joinseeds.com/letmein?client=seedslight'),
                 ),
+                // TextSpan(
```
is this needed?

gguijarro-c-chwy commented on lib/screens/onboarding/join_process/screens/show_onboarding_choice.dart:
```diff
@@ -114,15 +120,27 @@ class ShowOnboardingChoice extends StatelessWidget {
+        Container(
+          height: 10,
+        ),
+        // buildBottom()
```
needed?

gguijarro-c-chwy reviewed (changes requested):
This is a massive PR. 36 Files and 1733 - 841 changes.
It's honestly impossible for anyone to jump into this PR and property review it. 


This could have been split into multiple smaller more manageable PRs. 

I requested some changes for things that looked weird.

n13 committed:
use account name and private key from settings

This fixes various bugs where account name and private key could get out of sync or be null - likely timing issues. This way we always know there's only one source of truth for these variables, that's the settings.

7flash committed:
pass arguments to handlers

7flash committed:
Merge remote-tracking branch 'origin/feature/guardians-recovery' into feature/guardians-recovery

7flash committed:
remove back button on continue recovery - add cancel button
